src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -150,6 +157,10 @@ def spring_cloud_update(cmd, client, resource_group, name, app_insights_key=None
     _update_application_insights_asc_update(cmd, resource_group, name, location,
                                             app_insights_key, app_insights, disable_app_insights, no_wait)
 
+    if build_pool_size is not None:
+        _update_default_build_agent_pool(
```

Comment:
how about moving the `if build_pool_size is not None` inside `_update_default_build_agent_pool`

Reply:
OK